### PR TITLE
Replace success boolean by status code

### PR DESCRIPTION
**Pull request by desoindx**

### CODE REVIEW

File: test/test-appointmentsController.js [hunk 1]
```diff
@@ -296,7 +296,7 @@ describe('appointmentsController', () => {
         .set('Cookie', `token=${cookie.getJwtTokenForUser(psy.dossierNumber, 'randomXSRFToken')}`)
         .set('xsrf-token', 'randomXSRFToken')
         .then(async (res) => {
-          res.body.success.should.equal(false);
+          res.status.should.equal(400);
```
Comment (sandrica89) [inline test/test-appointmentsController.js:299]: Maybe 404 Not found?

File: test/test-patientsController.js [hunk 2]
```diff
@@ -424,7 +424,7 @@ describe('patientsController', () => {
         .set('Cookie', `token=${cookie.getJwtTokenForUser(psy.dossierNumber, 'randomXSRFToken')}`)
         .set('xsrf-token', 'randomXSRFToken')
         .then(async (res) => {
-          res.body.success.should.equal(false);
+          res.status.should.equal(400);
```
Comment (sandrica89) [inline test/test-patientsController.js:427]: I think we should return same http status as previous test. The user will therefore not know whether the patient does not exists or is not his/hers. I would put 404 here.

File: test/test-patientsController.js [hunk 3]
```diff
@@ -874,7 +874,7 @@ describe('patientsController', () => {
         .set('Cookie', `token=${cookie.getJwtTokenForUser(psy.dossierNumber, 'randomXSRFToken')}`)
         .set('xsrf-token', 'randomXSRFToken')
         .then(async (res) => {
-          res.body.success.should.equal(false);
+          res.status.should.equal(500);
```
Comment (sandrica89) [inline test/test-patientsController.js:877]: Maybe 404 Not found ?

File: test/test-patientsController.js [hunk 4]
```diff
@@ -896,7 +896,7 @@ describe('patientsController', () => {
       .set('Cookie', `token=${cookie.getJwtTokenForUser(psy.dossierNumber, 'randomXSRFToken')}`)
       .set('xsrf-token', 'randomXSRFToken')
         .then(async (res) => {
-          res.body.success.should.equal(false);
+          res.status.should.equal(400);
```
Comment (sandrica89) [inline test/test-patientsController.js:899]: Maybe 404 Not found? (No difference between invalid and not linked to the user)

File: frontend/src/services/agent.js [hunk 5]
```diff
@@ -23,16 +23,10 @@ const client = createClient();
...
-      store.commonStore.setNotification(response.data);
-      throw new Error(response.data.message);
-    }
-    return response.data;
-  },
+  response => response.data,
   error => {
     if (error.response) {
-      store.commonStore.setNotification(error.response.data);
+      store.commonStore.setNotification(error.response.data, false);
```
Comment (sandrica89) [inline frontend/src/services/agent.js:29]: Won't we display server error (real errors not `CustomError`) here? I didn't saw anything catching all unexpected errors to display a nice message like "Oops, something failed" but we may have already done it.

**Review by sandrica89 — CHANGES_REQUESTED** [review]

File: test/test-patientsController.js (hunk 2 above)
Comment (desoindx) [reply test/test-patientsController.js:427]: this is a bit different, the uuid validation is done by the type validator => 400

File: test/test-patientsController.js (hunk 4 above)
Comment (desoindx) [reply test/test-patientsController.js:899]: same => invalid uuid

File: frontend/src/services/agent.js (hunk 5 above)
Comment (desoindx) [reply frontend/src/services/agent.js:29]: we have a filter in the errorManager middleware that takes care of that
it displays the 500 error in dev but a generic message in prod

**Comment by sandrica89** [comment]
The page to add a new appointment is broken: https://sante-psy-etudiant-staging-pr309.osc-fr1.scalingo.io/psychologue/nouvelle-seance
Some changes must be missing on `NewAppointment.jsx`.

**Review by sandrica89 — APPROVED** [review]

**Pull request merged**